### ISSUE TITLE
Fine Tuning - not improve metrics

### ISSUE THREAD

**Issue by jesusbft** [mittagessen/party#9]
Trying fine tuning the party pretrained model for the Portuguese and I didn't get any improvements.

The parameters:

party -d cuda:0 --precision bf16-mixed \
train -q early --lag 20 --augment \
--optimizer AdamW --freeze-encoder \
--warmup 5000 -r 0.0003 -w 0.0001 --schedule constant \
-B 32 --workers 16 --threads 16 \
-t dataset/train.lst -e dataset/val.lst \
-o models/Portuguese

Epoch 0/-2 ━━━━6253/6253 1:13:14 • 0:00:00 1.61it/s v_num: 3.000 train_loss: 0.070 val_metric: 0.245
Epoch 1/-2 ━━━━0/6253 0:00:00 • -:--:-- 0.00it/s  Trainer was signaled to stop but the required `min_epochs=10` or `min_steps=None` has not been met. Training will continue...
Epoch 1/-2 ━━━━6253/6253 1:05:38 • 0:00:00 1.59it/s v_num: 3.000 train_loss: 0.654 val_metric: 0.234
Epoch 2/-2 ━━━━6253/6253 1:05:44 • 0:00:00 1.59it/s v_num: 3.000 train_loss: 0.131 val_metric: 0.246
Epoch 3/-2 ━━━━6253/6253 1:06:33 • 0:00:00 1.58it/s v_num: 3.000 train_loss: 0.203 val_metric: 0.242
Epoch 4/-2 ━━━━6253/6253 1:05:38 • 0:00:00 1.59it/s v_num: 3.000 train_loss: 0.228 val_metric: 0.239
Epoch 5/-2 ━━━━6253/6253 1:05:37 • 0:00:00 1.59it/s v_num: 3.000 train_loss: 0.115 val_metric: 0.223
Epoch 6/-2 ━━━━6253/6253 1:05:45 • 0:00:00 1.59it/s v_num: 3.000 train_loss: 0.212 val_metric: 0.241

Any ideia?

**Comment by mittagessen** [comment]
Yes, I pretrained the model but I've been trying unsuccessfully for a while now to get
it to fine-tune on any size of dataset. No recommendations yet but I'll
hope to get some input from a colleague next week that will allows me to
make some progress on that front.

**Comment by jesusbft** [comment]
Thank you for letting me know, I'll wait for updates.

By the way, is possible to train from scratch for tests? I checked the code and the default is load from pretrained model. If I ignore this code, will the command work?

    if load_from_checkpoint and load_from_repo:
        raise click.BadOptionsUsage('load_from_checkpoint', 'load_from_* options are mutually exclusive.')
    elif load_from_checkpoint is None and load_from_repo is None:
        load_from_repo = '10.5281/zenodo.14616981'

**Comment by mittagessen** [comment]
On 25/02/02 04:24PM, Weslley Oliveira wrote:
> Thank you for letting me know, I'll wait for updates.
> 
> By the way, is possible to train from scratch for tests? I checked the
> code and the default is load from pretrained model. If I ignore this
> code, will the command work?

This snippet should work to produce a checkpoint that only contains the
pretrained image encoder + ByteLlama language model:

```

from party.model import RecognitionModel
from lightning import Trainer
model = RecognitionModel()
trainer = Trainer()
trainer.strategy.connect(model)
trainer.save_checkpoint('foo.ckpt')
```

It won't converge without a couple 100k pages worth of training data,
half a dozen A40s, and a week or two of compute though. Hence the
pretrained model. That should be fairly easy to fine-tune with small
datasets once I get it working.

**Comment by jesusbft** [comment]
I agree. Thank you!!

**Comment by mittagessen** [comment]
Fine-tuning works now and there's also a new base model.
